### PR TITLE
Merge nodejs/master into xplat

### PR DESCRIPTION
**Pull request by boingoing**
##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines][]

##### Affected core subsystem(s)

chakrashim

[commit guidelines]: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines

### CODE REVIEW

**Comment by boingoing** [comment]
Kicked-off a CI run:
https://ci.nodejs.org/job/chakracore-test-windows/56/

**Comment by digitalinfinity** [comment]
Pushed a commit to fix remaining test failures- @kfarnung @jasongin can you review please?

**Comment by kfarnung** [comment]
Kicked off CI:  ~~https://ci.nodejs.org/job/chakracore-test-windows/57/~~ https://ci.nodejs.org/job/chakracore-test-windows/58/

File: test/parallel/test-cli-syntax.js [hunk 1]
```diff
@@ -53,7 +53,12 @@ const syntaxArgs = [
     assert.strictEqual(c.stdout, '', 'stdout produced');
 
     // stderr should include the filename
-    assert(c.stderr.startsWith(file), "stderr doesn't start with the filename");
+    // (v8 prints the filename to the error stream- chakracore does not)
+    console.log(c.stderr);
```
Comment (kfarnung) [inline test/parallel/test-cli-syntax.js]: Used for debugging?  Should we remove it?

**Review by kfarnung — APPROVED** [review]
LGTM 🚢 pending CI run in progress.

File: test/parallel/test-cli-syntax.js (hunk 1 above)
Comment (digitalinfinity) [reply test/parallel/test-cli-syntax.js]: gah- good catch- pushed a commit to remove it.

**Comment by digitalinfinity** [comment]
@kfarnung thanks for the review- do you have a link to the CI run?

**Comment by kfarnung** [comment]
@digitalinfinity CI link was called out above:

https://ci.nodejs.org/job/chakracore-test-windows/58/

Looks like there's one failure for 2012r2 only, most likely a flaky test.  Will kick off one last CI run to see if we can get a clean test pass.

**Comment by kfarnung** [comment]
New CI:  https://ci.nodejs.org/job/chakracore-test-windows/59/

**Comment by digitalinfinity** [comment]
Landed in commits 481ca0e, 361c4bb, ac1e9e0

**Pull request closed without merging**